staging/src/kubevirt.io/api/export/v1alpha1/types.go

```diff
@@ -104,6 +104,12 @@ type VirtualMachineExportStatus struct {
 	// create the internal URLs for downloading the images
 	ServiceName string `json:"serviceName,omitempty"`
 
+	// +optional
+	// VirtualMachineName shows the name of the source virtual machine if the source is either a VirtualMachine or
+	// a VirtualMachineSnapshot. This is mainly to easily identify the source VirtualMachine in case of a
+	// VirtualMachineSnapshot
+	VirtualMachineName string `json:"virtualMachineName,omitempty"`
```

Comment:
I think this type should probably be `*string`.  Will be unset in the case of PVC source

https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#optional-vs-required

Reply:
Good point, I will make that change.

Reply:
Does that doc need to be updated as `+optional` and `omitempty` work fine with empty `string` values? I get that using a pointer lets us check for unset values but we'd expect the name here to be empty or set, never unset right?

Anyway we should really have some kind of lint rule somewhere to enforce the use of pointers for strings if we want things to be consistent, just above this `ServiceName` is also optional and not a pointer.

Reply:
So with just a string we overload `""` as does not exist, even though blank might be a valid value. With *string we can use nil as does not exist, and blank is just another possible value. The difference is mostly academic, but in general the rule is, if you have an optional field, it should be a pointer.

Reply:
Yeah understood, my point was that "" isn't a valid name for a VirtualMachine so we could use it to mean unset here. Either way using pointers for optional attributes is definitely something we can check and enforce in the codebase but that's for another PR, apologies for the noise!

Reply:
@lyarwood good point, `ServiceName` maybe shouldn't be optional.  There will always be a service associated to a VMExport and it's name is known as soon as the VMExport is created

What makes `VirtualMachineName` different from `ServiceName` is that `VirtualMachineName` will only be set for certain VMExports, not all